### PR TITLE
Add timer and retry button in phone verification screen

### PR DESCRIPTION
**Pull request by maxghafori**
[issue 210](https://github.com/GaloyMoney/galoy-mobile/issues/210)

Suggest retrying send text after 60 sec in phone verification screen.

### CODE REVIEW

File: app/components/timer/utils.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export function parseTimer(seconds: number): string {
+  if (!seconds) {
+    return "00:00"
+  }
+  const min = parseInt(String(seconds / 60)).toString()
+  const sec = parseInt(String(seconds % 60)).toString()
+  const minute = parseInt(min) < 10 ? `0${min}` : min
```
Comment (jrbuhl93) [inline app/components/timer/utils.ts:7]: Cleaner to use [padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart) here.

File: app/screens/phone-auth-screen/phone-auth.tsx [hunk 2]
```diff
@@ -317,6 +347,24 @@ export const WelcomePhoneValidationScreen: ScreenType = ({
...
+              >
+                <Text
+                  style={{
+                    color: timer === 0 ? color.palette.black : color.palette.midGrey,
+                  }}
+                >
+                  {translate("WelcomePhoneValidationScreen.sendAgain")}
+                </Text>
+              </Pressable>
+              <Timer onTimerFinish={() => setTimer(0)} count={timer} />
```
Comment (jrbuhl93) [inline app/screens/phone-auth-screen/phone-auth.tsx:366]: I think hiding the timer might look good once it reaches 0.

**Review by jrbuhl93 — COMMENTED** [review]
Let's also remove the PhoneBadger component on line 325. There isn't enough space on most screen size for it, and the image will cause the keyboard to cover up the timer so users won't know it is there.

This is a great pr! Thank you.

File: app/screens/phone-auth-screen/phone-auth.tsx (hunk 2 above)
Comment (maxghafori) [reply app/screens/phone-auth-screen/phone-auth.tsx:366]: I changed it to show timer when it's more than 0 and also show `Send Again` button when timer is 0.

**Comment by maxghafori** [comment]
Thank you for your comments , I fixed them.

**Comment by jrbuhl93** [comment]
> Thank you for your comments , I fixed them.

Looks like you might have forgotten to add the changes to your new commit.

**Comment by maxghafori** [comment]
Sorry for that, now it should be ok .

**Comment by maxghafori** [comment]
I couldn't use alias path `@app` for import component because i got error.

If you agree, I will add this plugin to babel config 
`plugins: [['module-resolver',{ alias: {'@app': './app'}}]]`

and also I should add this dev dependency
`yarn add --dev babel-plugin-module-resolver`

**Comment by jrbuhl93** [comment]
> I couldn't use alias path `@app` for import component because i got error.
> 
> If you agree, I will add this plugin to babel config `plugins: [['module-resolver',{ alias: {'@app': './app'}}]]`
> 
> and also I should add this dev dependency `yarn add --dev babel-plugin-module-resolver`

You are right that for now we aren't set up to resolve @app (except in the tests folder). Hold off on adding the plugin for now. Samer is going to look into alternatives.

File: app/components/timer/utils.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+export function parseTimer(seconds: number): string {
```
Comment (jrbuhl93) [inline app/components/timer/utils.ts:1]: Could you add some unit tests for parseTimer?

File: app/components/timer/utils.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  if (!seconds) {
+    return "00:00"
+  }
+  const minute = parseInt(String(seconds / 60)).toString()
+  const second = parseInt(String(seconds % 60)).toString()
+  return `${minute.padStart(2, '0')}:${second.padStart(2, '0')}`
+}
+
+// this function is used to compute correct time spent
+export function computeCount(startTime: Date): number {
```
Comment (jrbuhl93) [inline app/components/timer/utils.ts:11]: And unit tests for computeCount

**Review by jrbuhl93 — COMMENTED** [review]
Thanks for making the change to hide the timer at 0. But I think we should keep the disabled "send again" while the timer is active, so that users know the purpose of the timer and aren't confused.

**Comment by maxghafori** [comment]
Now the Send Again button is shown when disabled.

**Pull request merged**